### PR TITLE
Mic sample fix

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change fixes the Windows Mic Stream demo when imported using UPM packages by creating a new assembly definition for the application and referencing / checking for the com.microsoft.mixedreality.toolkit.micstream package.

Previous references in the root examples assembly definition have been removed.

## Changes
- Fixes: #9908

## Verification
Side-loaded test packages, imported Demos - Audio, deployed to HoloLens 2, talked quietly and loudly :smile:

### CODE REVIEW

File: scripts/ci/validatecode.ps1 [hunk 1]
```diff
@@ -586,6 +586,7 @@ $AsmDefExceptions = [System.Collections.Generic.HashSet[String]]@(
     "Assets/MRTK/Core/Utilities/Gltf/MRTK.Gltf.asmdef",
     "Assets/MRTK/Core/Utilities/Gltf/Serialization/Importers/MRTK.Gltf.Importers.asmdef",
     "Assets/MRTK/Examples/MRTK.Examples.asmdef",
+    "Assets/MRTK/Examples/Demos/Gltf/MRTK.Demos.Audio.asmdef",
```
Comment (keveleigh) [inline scripts/ci/validatecode.ps1:589]:
```suggestion
    "Assets/MRTK/Examples/Demos/Audio/MRTK.Demos.Audio.asmdef",
```

**Review by MaxWang-MS — APPROVED** [review]

File: scripts/ci/validatecode.ps1 (hunk 1 above)
Comment (davidkline-ms) [reply scripts/ci/validatecode.ps1]: thx for the catch!

**Pull request merged**